benchmarks/src/bin/tpch.rs

```diff
@@ -618,291 +630,412 @@ mod tests {
     }
 
-    #[cfg(feature = "ci")]
     #[tokio::test]
-    async fn verify_q12() -> Result<()> {
-        verify_query(12).await
+    async fn run_q12() -> Result<()> {
+        run_query(12).await
     }
 
-    #[cfg(feature = "ci")]
     #[tokio::test]
-    async fn verify_q13() -> Result<()> {
-        verify_query(13).await
+    async fn run_q13() -> Result<()> {
+        run_query(13).await
     }
 
-    #[cfg(feature = "ci")]
     #[tokio::test]
-    async fn verify_q14() -> Result<()> {
-        verify_query(14).await
+    async fn run_q14() -> Result<()> {
+        run_query(14).await
     }
 
-    #[cfg(feature = "ci")]
     #[tokio::test]
-    async fn verify_q15() -> Result<()> {
-        verify_query(15).await
+    async fn run_q15() -> Result<()> {
+        run_query(15).await
     }
 
-    #[cfg(feature = "ci")]
     #[tokio::test]
-    async fn verify_q16() -> Result<()> {
-        verify_query(16).await
+    async fn run_q16() -> Result<()> {
+        run_query(16).await
     }
 
-    #[cfg(feature = "ci")]
     #[tokio::test]
-    async fn verify_q17() -> Result<()> {
-        verify_query(17).await
+    async fn run_q17() -> Result<()> {
+        run_query(17).await
     }
 
-    #[cfg(feature = "ci")]
     #[tokio::test]
-    async fn verify_q18() -> Result<()> {
-        verify_query(18).await
+    async fn run_q18() -> Result<()> {
+        run_query(18).await
     }
 
-    #[cfg(feature = "ci")]
     #[tokio::test]
-    async fn verify_q19() -> Result<()> {
-        verify_query(19).await
+    async fn run_q19() -> Result<()> {
+        run_query(19).await
     }
 
-    #[cfg(feature = "ci")]
     #[tokio::test]
-    async fn verify_q20() -> Result<()> {
-        verify_query(20).await
+    async fn run_q20() -> Result<()> {
+        run_query(20).await
     }
 
-    #[cfg(feature = "ci")]
     #[tokio::test]
-    async fn verify_q21() -> Result<()> {
-        verify_query(21).await
+    async fn run_q21() -> Result<()> {
+        run_query(21).await
     }
 
-    #[cfg(feature = "ci")]
     #[tokio::test]
-    async fn verify_q22() -> Result<()> {
-        verify_query(22).await
+    async fn run_q22() -> Result<()> {
+        run_query(22).await
+    }
+
+    async fn run_query(n: usize) -> Result<()> {
+        // Tests running query with empty tables, to see whether they run successfully.
+
+        let config = SessionConfig::new()
+            .with_target_partitions(1)
+            .with_batch_size(10);
+        let ctx = SessionContext::with_config(config);
+
+        for &table in TPCH_TABLES {
+            let schema = get_tpch_table_schema(table);
+            let batch = RecordBatch::new_empty(Arc::new(schema.to_owned()));
+
+            ctx.register_batch(table, batch)?;
+        }
+
+        let sql = &get_query_sql(n)?;
+        for query in sql {
+            execute_query(&ctx, query, false).await?;
+        }
+
+        Ok(())
+    }
+}
+
+/// CI checks
+#[cfg(test)]
+#[cfg(feature = "ci")]
```

Comment:
Why feature gate this with `ci`? I would have thought

```rust
#[cfg(test)]
```

was sufficient. 
 
Additionally, I couldn't find evidence that these new tests were actually run:


```shell
-*- mode: grep; default-directory: "~/Downloads/logs_52324/" -*-
Grep started at Tue Nov  1 12:19:52

rg -n -H --no-heading -e 'serde_q7' 

Grep finished with no matches found at Tue Nov  1 12:19:52
```

🤔

Reply:
These tests rely on the data files existing in a path from `TPCH_DATA` env var, which is set in CI

Reply:
I did look into using MemTable but we don't support serde with MemTable